lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -185,28 +185,31 @@ def package_manager
       end
 
       def os_facts
-        facter = ForemanMaintain::Utils::Facter.path
-        @os_facts ||= JSON.parse(execute("#{facter} -j os"))
+        @os_facts ||= ForemanMaintain::Utils::OsFacts
       end
 
       def el?
-        os_facts['os']['family'] == 'RedHat'
+        ids = ['rhel', 'rhel fedora']
+        ids.include?(os_facts.id_like) ||
+          ids.include?(os_facts.id)
```

Comment:
This won't work on  some EL-derivatives. https://github.com/stejskalleos/os_release is a nice collection of the various os-release files if you want examples.

Another way is to check for the existence of `/etc/redhat-release`.

Reply:
I think checking for `/etc/redhat-release` is the most trivial and correct way.

This *will* flag Fedora as being EL, but so does the current code?
And we don't really care about Fedora right now.

Reply:
and same for this one, if in  case redhat-release file is missing we check for os-release.